drakrun/drakrun/storage.py

```diff
@@ -308,9 +310,247 @@ def import_vm0(self, path: str):
         shutil.copy(path, os.path.join(VOLUME_DIR, 'vm-0.img'))
 
 
+class LvmStorageBackend(StorageBackendBase):
+    """Implements storage backend based on lvm storage"""
+
+    def __init__(self, install_info: InstallInfo):
+        super().__init__(install_info)
+        self.lvm_volume_group = install_info.lvm_volume_group
+        self.install_info = install_info
+        self.check_tools()
+
+    def check_tools(self):
+        """ Verify existence of lvm command utility """
+        try:
+            subprocess.run(["vgs", self.lvm_volume_group], check=True, stdout=subprocess.DEVNULL)
+        except subprocess.CalledProcessError:
+            raise RuntimeError("Failed to execute vgs command"
+                               f"Make sure you have LVM support installed with {self.lvm_volume_group} as a volume group")
+
+    def initialize_vm0_volume(self, disk_size: str):
+        """Create base volume for vm-0 with given size
+
+        disk_size - string representing volume size with M/G/T suffix, eg. 100G
+        """
+        try:
+            logging.info("Deleting existing logical volume and snapshot")
+            subprocess.run(
+                ' '.join([
+                    "lvremove",
+                    "-v",
+                    "-y",
+                    # "--noudevsync",
+                    f"{self.lvm_volume_group}/vm-0-snap"
+                ]),
+                shell=True
+            )
+            subprocess.check_output(
+                ' '.join([
+                    "lvremove",
+                    "-v",
+                    "-y",
+                    # "--noudevsync",
+                    f"{self.lvm_volume_group}/vm-0"
+                ]),
```

Comment:
Don't pass `shell=True` and you will be then able to pass the command line as an array without need to join it manually.

Reply:
Sure, will do it. I just prefer `shell=True` as it is more close the shell manipulations and doesn't introduce much bugs which I have sometimes faced in arrays when doing shell manipulations with `|` and all